lib/pure/collections/sequtils.nim

```diff
@@ -895,34 +873,14 @@ template mapIt*(s: typed, op: untyped): untyped =
       strings = nums.mapIt($(4 * it))
     assert strings == @["4", "8", "12", "16"]
 
-  when defined(nimHasTypeof):
-    type OutType = typeof((
-      block:
-        var it{.inject.}: typeof(items(s), typeOfIter);
-        op), typeOfProc)
-  else:
-    type OutType = type((
-      block:
-        var it{.inject.}: type(items(s));
-        op))
-  when compiles(s.len):
-    block: # using a block avoids https://github.com/nim-lang/Nim/issues/8580
-
-      # BUG: `evalOnceAs(s2, s, false)` would lead to C compile errors
-      # (`error: use of undeclared identifier`) instead of Nim compile errors
-      evalOnceAs(s2, s, compiles((let _ = s)))
-
-      var i = 0
-      var result = newSeq[OutType](s2.len)
```

Comment:
Ah so your version is simpler but misses an important optimization entirely!

Reply:
This so called "import optimization" comes at the cost of a full deep copy of the input parameter. This deep copy can be very expensive. The question me was to either have a questionable optimization (how I would call it) with a complex and hard to read implementation, or a straight forward implementation with well understood performance characteristics. I picked the straight forward implementation.